src/parser/jobs/nin/Ninjutsu.js

```diff
@@ -30,8 +39,54 @@ export default class Ninjutsu extends Module {
 		this._rabbitCount++
 	}
 
+	_onDotonCast(event) {
+		this._dotonCasts.push({
+			cast: event,
+			tcj: this.combatants.selected.hasStatus(STATUSES.TEN_CHI_JIN.id), // STDs are only okay under TCJ
+			ticks: [],
+		})
+	}
+
+	_onDotonDamage(event) {
+		// If there are no casts at all, use the damage event to fabricate one
+		if (this._dotonCasts.length === 0) {
+			this._onDotonCast(event)
+		}
+
+		this._dotonCasts[this._dotonCasts.length - 1].ticks.push(event.hits.length) // Track the number of enemies hit per tick
+	}
+
+	_appraiseDotonCasts() {
+		const result = {
+			badTcjs: 0, // TCJ Dotons that fell short of 7 ticks (TCJ Suiton will be more damage)
+			badAoes: 0, // AoE Dotons that fell short of 6 ticks (Katon will be more damage)
+			badStds: 0, // Single-target regular Dotons (do not do this)
+		}
+
+		for (let i = 0; i < this._dotonCasts.length; i++) {
```

Comment:
`Array.prototype.forEach` is a friendo. This will work fine, though.

Reply:
I'll start using this instead, gotta get over my trust issues one of these days.